tensorflow/compiler/xla/mlir_hlo/lib/Dialect/mhlo/IR/hlo_ops.cc

```diff
@@ -8871,6 +8875,47 @@ LogicalResult UniformDequantizeOp::inferReturnTypeComponents(
   return success();
 }
 
+//===----------------------------------------------------------------------===//
+// DynamicConvOp
+//===----------------------------------------------------------------------===//
+
+// simplify dynamic_conv = > convolution
+static LogicalResult simplifyDynamicConv(DynamicConvOp op,
+                                         PatternRewriter &rewriter) {
+  DenseIntElementsAttr dPaddingAttr;
+  if (!matchPattern(op.getDPadding(), m_Constant(&dPaddingAttr))) {
+    return failure();
+  }
+  size_t spatialDim = op.dimension_numbers().getInputSpatialDimensions().size();
+  assert(dPaddingAttr.size() == spatialDim * 2);
+
+  llvm::SmallVector<int64_t> newPadding = llvm::to_vector(
+      llvm::map_range(dPaddingAttr.getValues<APInt>(),
+                      [&](APInt i) { return i.getSExtValue(); }));
+  if (op.getPadding().hasValue()) {
```

Comment:
Return a match error if there is a padding value that differs. If there is a disagreement between the attribute and the passed value somethign is wrong.

Reply:
I removed `assert()`, please review again, thanks!